app/models/library_entry.rb

```diff
@@ -209,6 +209,21 @@ def kind
       user.update_feed_completed!
       user.update_profile_completed!
     end
+
+    if progress_changed?
+      case kind
+      when :anime
+        guess = [(progress + 1), anime.default_progress_limit].min
+        anime.update_episode_count_guess(guess) if
+          !anime.episode_count &&
+          (!anime.episode_count_guess || anime.episode_count_guess <= guess)
```

Comment:
Does this conditional logic really belong in LibraryEntry or would it make more sense in the media?